domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java

```diff
@@ -439,4 +498,137 @@ public static void deleteById(String table, String id,
   public static PostgresClient postgresClient(Context vertxContext, Map<String, String> okapiHeaders) {
     return PostgresClient.getInstance(vertxContext.owner(), TenantTool.tenantId(okapiHeaders));
   }
+  /**
+   * Generate a CQLWrapper object for use by optimization.
+   *
+   * @param query the cql query to parse
+   * @param limit the max objects to return
+   * @param offset the start index of objects to return
+   * @param fields the fields to apply the cql query above to
+   * @return CQLWrapper
+   */
+  public static CQLWrapper createCQLWrapper(
+    String query,
+    int limit,
+    int offset,
+    List<String> fields) throws FieldException {
+
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON(fields);
+
+    return new CQLWrapper(cql2pgJson, query)
+      .setLimit(new Limit(limit))
+      .setOffset(new Offset(offset));
+  }
+  public static class SQLOptimizationException extends Exception {
+    private static final long serialVersionUID = 7772172153233445467L;
+    public SQLOptimizationException(String string) {
+      super(string);
+    }
+  }
+  /**
+   * Generate optimized sql given a specific cql query, tenant, index column name hint and configurable size to hinge the optimization on.
+   *
+   * @param preparedCql the cql query
+   * @param tenantId the tenant used to generate schema location
+   * @param offset start index of objects to return
+   * @param limit max number of objects to return
+   * @param column the index column to use 
+   * @param size the number of rows that determines which method will be used to generate the ultimate result 
+   * @return CQLWrapper
+   * @throws Exception on generation failure
+   */
+  public static String optimizedSql(PreparedCQL preparedCql, String tenantId, 
```

Comment:
Line 537 says "@return CQLWrapper" but line 540 says: "public static String optimizedSql("

The class CQLWrapper and the class String are different and incompatible.

The javadoc about the return type of method optimizedSql in line 537 contradicts the actual return type declared in line 430.

Link to these two lines:
https://github.com/folio-org/raml-module-builder/blob/acdea7941d353f986e06608d7550d4f83ac42da3/domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java#L537
https://github.com/folio-org/raml-module-builder/blob/acdea7941d353f986e06608d7550d4f83ac42da3/domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java#L540